src/edge_index.cpp

```diff
@@ -106,6 +111,127 @@ Selection resolve(const HighFive::Group& indexGroup, const std::vector<NodeID>&
     return Selection::fromValues(result);
 }
 
+
+namespace {
+
+std::unordered_map<NodeID, RawIndex> _groupNodeRanges(const std::vector<NodeID>& nodeIDs)
+{
+    std::unordered_map<NodeID, RawIndex> result;
+
+    if (nodeIDs.empty()) {
+        return result;
+    }
+
+    uint64_t i0 = 0;
+    NodeID lastNodeID = nodeIDs[0];
```

Comment:
nit: `nodeIDs[id0]`, to show what i0 is used for.  I'd consider renaming `i0` `group_start_idx` or something, personally

Reply:
OK